rpc/cc/arduino/cli/commands/v1/upload.proto

```diff
@@ -52,6 +53,12 @@ message UploadRequest {
   // If set to true, the actual upload will not be performed but a trace output
   // will be printed stdout. This is for debugging purposes.
   bool dry_run = 10;
+  // User provided fields usually used by upload tools that need authentication
+  // or in any case fields that can be customized by the user at upload time
+  // and cannot be known previously.
+  // For more info:
+  // https://github.com/arduino/tooling-rfcs/blob/main/RFCs/0002-pluggable-discovery.md#user-provided-fields
```

Comment:
```suggestion
  // https://github.com/arduino/tooling-rfcs/blob/main/RFCs/0002-pluggable-discovery.md#user-provided-fields
```
I wonder if it makes more sense to reference the platform specification instead of the RFC in this sort of comment?

It seems that you have documented most of the important information from the RFC in the CLI docs specifications now, so it seems like the RFC can serve more as a decision record while the specifications are the actively maintained documentation that is updated as the system evolves over time.

Reply:
This kind of a vicious circle really, when I wrote that comment I still didn't write the documentation so I had only the RFC available.
Also I think there are other occurences in this PR where I do the same.

Reply:
>I think there are other occurences in this PR where I do the same.

Correct. I picked on at random to comment on to see what your opinion was.

What do you think? Is the specification a better reference to direct the reader to, or the RFC?

Reply:
Yeah, the specification is probably the best thing. I'll update the links, hope I'll get them correctly.

Reply:
Done: https://github.com/arduino/arduino-cli/commit/b59b6f6a43835760ee056d7fedd0c71611c640c7

Reply:
Looks great. Thanks!